src/overlays/actors/ovl_En_Wdhand/z_en_wdhand.c

```diff
@@ -105,63 +175,679 @@ static DamageTable D_80AF64F4 = {
+                    EffectSsDtBubble_SpawnColorProfile(play, &effPos, &effectVelocity, &effectAccel, effectScale, 25, 2,
+                                                       true);
+                }
+            }
+
+            Actor_Kill(&this->actor);
+        } else {
+            this->actor.scale.z = this->actor.scale.y = this->actor.scale.x;
+        }
+    }
+
+    for (i = 0; i < ARRAY_COUNT(this->endPoints); i++) {
+        effectVelocity.y = Rand_ZeroOne() + 1.0f;
+        EffectSsDtBubble_SpawnColorProfile(play, &this->endPoints[i], &effectVelocity, &effectAccel,
+                                           Rand_S16Offset(40, 40), 25, 2, true);
+    }
+}
+
+void EnWdhand_UpdateDamage(EnWdhand* this, PlayState* play) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        Player* player = GET_PLAYER(play);
+
+        this->collider.base.acFlags &= ~AT_ON;
+        this->collider.base.acFlags &= ~AC_HIT;
+        this->collider.base.atFlags &= ~AT_ON;
+        this->collider.base.ocFlags1 &= ~AT_ON;
+
+        Actor_SetDropFlagJntSph(&this->actor, &this->collider);
+        Enemy_StartFinishingBlow(play, &this->actor);
+
+        if ((player->stateFlags2 & PLAYER_STATE2_80) && (&this->actor == player->actor.parent)) {
+            // Drop the player
+            player->av2.actionVar2 = 100;
+            player->actor.parent = NULL;
+            player->actor.shape.rot.x = 0;
+            player->actor.shape.rot.z = 0;
+            func_800B8D50(play, &this->actor, this->actor.speed, this->actor.world.rot.y, this->actor.velocity.y, 0);
+        } else {
+            // Only transpose if player is not grabbed, since it is already in the correct state if so.
+            Matrix_Transpose(&this->relativeToWorldTransform);
+        }
+
+        EnWdhand_SetupDie(this);
+    }
+}
+
+void EnWdhand_Update(Actor* thisx, PlayState* play) {
+    EnWdhand* this = THIS;
+
+    EnWdhand_UpdateDamage(this, play);
+
+    this->actionFunc(this, play);
+
+    if (this->collider.base.atFlags & AT_ON) {
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->collider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->collider.base.ocFlags1 & OC1_ON) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
+
+void EnWdhand_UpdateColliderLocationsForLimb(EnWdhand* this, s32 limbIndex, Vec3f* limbPos) {
+    EnWdhand_SetTransform(this, limbIndex, true, limbPos);
+    Matrix_MultVecY(575.0f, limbPos);
+    EnWdhand_Vec3fToVec3s(&this->collider.elements[2 * limbIndex + 0].dim.worldSphere.center, limbPos);
+    Matrix_MultVecY(1725.0f, limbPos);
+    EnWdhand_Vec3fToVec3s(&this->collider.elements[2 * limbIndex + 1].dim.worldSphere.center, limbPos);
+    Matrix_MultVecY(2300.0f, limbPos);
+}
+
+void EnWdhand_Draw(Actor* thisx, PlayState* play) {
+    EnWdhand* this = THIS;
+    Vec3f limbPos;
+    Gfx* dl;
```

Comment:
```suggestion
    Gfx* gfx;
```
Lately I think I have been going with `gfx` over `dl`.